drivers/ina3221/alerts.c

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2019 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_ina3221
+ * @{
+ *
+ * @file
+ * @brief       Functions to enable/disable GPIO alerts
+ *              for INA3221
+ *
+ * @author      Fabian Hüßler <fabian.huessler@ovgu.de>
+ *
+ * @}
+ */
+
+#include <errno.h>
+#include <periph/gpio.h>
+#include "ina3221.h"
+
+int _ina3221_enable_alert(ina3221_t *dev, ina3221_alert_t alert,
+                          ina3221_alert_cb_t cb, void *arg)
+{
+    if (alert >= INA3221_NUM_ALERTS) {
+        return -ERANGE;
+    }
+    if (dev->params.alert_pins[alert] == GPIO_UNDEF) {
+        return -ENOTSUP;
+    }
+    dev->alert_callbacks[alert] = cb;
+    dev->alert_callback_arguments[alert] = arg;
+    int check = gpio_init_int(
+        dev->params.alert_pins[alert],
+        GPIO_IN,
```

Comment:
You could use `GPIO_IN_PU` over `GPIO_IN` to use the MCU provided pull up resistor, [see doc](http://api.riot-os.org/cc2538_2include_2periph__cpu_8h.html#a491a2cbfb4e94f2afcc0d5bdef2dc454). It would however make sense to allow the user to still use an external pull up (and use `GPIO_IN` rather than `GPIO_IN_PU`), as not every MCU has internal pull ups on every pin. This could be done via the `ina3221_params`, or also as global define.

(The global define would result in a slightly small RAM/ROM requirements. But if one would use multiple INA3221 device on the same MCU, with some being connected to pins with internal pull up and some without, the global defines would force the use to use external pull ups in any case, while the `ina3221_params` version would not.)